lib/merit/controller_extensions.rb

```diff
@@ -46,16 +46,20 @@ def had_errors?
       target_object.respond_to?(:errors) && target_object.errors.try(:present?)
     end
 
-    def target_object
-      target_obj = instance_variable_get(:"@#{controller_name.singularize}")
-      if target_obj.nil?
+    def warn_no_object_found
         str = '[merit] No object found, you might need a ' \
           "'@#{controller_name.singularize}' variable in " \
           "'#{controller_path}_controller' if no reputation is applied. " \
           'If you are using `model_name` option in the rule this is ok.'
         Rails.logger.warn str
+    end
```

Comment:
It's useful to put the more relevant code at the top of the class, and the implementation details at the bottom. Also it's how code typically reads: `warn_no_object_found` is called from a line after its definition. For these two reasons, I'd switch the order of the two methods.